File: golang/1-two-sum.go
package golang

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	n := len(nums)
	for i := 0; i < n; i++ {
		complement, exists := m[nums[i]]
		if exists {
			return []int{i, complement}
		}
		m[target-nums[i]] = i
	}
	return []int{-1, -1}
}

File: php/1-two-sum.php
class Solution {
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $result = [];
        $hashMap = [];
        for ($i = 0; $ $i < count($nums); $i++) {
            $complement = $hashMap[$target - $nums[$i]];
            if ($complement !== null) {
                $result = [$i, $complement];
                break;
            }
            $hashMap[$nums[$i]] = $i;
        }
        return $result;
    }
}